packages/react/src/core/FileTree.tsx

```diff
@@ -231,7 +259,7 @@ function buildFileList(
 
       if (depth === segments.length - 1) {
         fileList.push({
-          kind: 'file',
+          kind: name.includes('.') ? 'file' : 'folder',
```

Comment:
Oh this is a bit hacky. I can see why you needed something like this given that now `files` no longer contains "just" files but can also contains folders. I think we should change the `files` prop from `string[]`  to `{ path: string, type: 'file' | 'folder' }[]`

Reply:
I also thought about this at first but didn't want to break existing integrations. But I'll mark this as `breaking change` and change the prop type.